ci/pkl-pipelines/pay-deploy/concourse-runner.pkl

```diff
@@ -12,26 +13,78 @@ resource_types {
 resources = new {
   pipeline_self_update.PayPipelineSelfUpdateResource("pay-deploy/concourse-runner.pkl", "master")
   shared_resources.payCiGitHubResource
-  shared_resources.payGithubPullRequestResource("concourse-runner-pr", "pay-ci")
-    |> withPaths(new Listing<String>{
-        "ci/docker/concourse-runner/**";
-        "ci/docker/concourse-runner-with-java-17/**";
-        "ci/docker/concourse-runner-with-java-21/**";
-        "ci/pipelines/concourse-runner.yml" })
-  shared_resources.payGithubResourceWithBranch("adminusers-master", "pay-adminusers", "master")
-  shared_resources.payGithubResourceWithBranch("webhooks-main", "pay-webhooks", "main")
-  shared_resources.payGithubResourceWithBranch("concourse-runner-src", "pay-ci", "master")
-    |> withPath("ci/docker/concourse-runner/*")
-  shared_resources.payGithubResourceWithBranch("concourse-runner-with-java-17-src", "pay-ci", "master")
-    |> withPath("ci/docker/concourse-runner-with-java-17/*")
-  shared_resources.payGithubResourceWithBranch("concourse-runner-with-java-21-src", "pay-ci", "master")
-  |> withPath("ci/docker/concourse-runner-with-java-21/*")
-  shared_resources.payDockerHubResource("concourse-runner", "governmentdigitalservice/pay-concourse-runner", "latest")
-  shared_resources.payDockerHubResource("concourse-runner-with-java-17", "governmentdigitalservice/pay-concourse-runner-with-java-17", "latest")
-  shared_resources.payDockerHubResource("concourse-runner-with-java-21", "governmentdigitalservice/pay-concourse-runner-with-java-21", "latest")
+  new PayResources.PayGitHubPullRequestResource {
```

Comment:
I think using functions here will reduce the boilerplate code `new ...{}` and keep the module small.

Reply:
Yeah, was wanting to demo how this worked so we can chat in the PKL retro.